istioctl/pkg/cli/context.go

```diff
@@ -103,6 +117,37 @@ func (i *instance) CLIClient() (kube.CLIClient, error) {
 	return i.CLIClientWithRevision("")
 }
 
+func (i *instance) CLIClientsForContexts(contexts []string) ([]kube.CLIClient, error) {
+	if i.remoteClients == nil {
+		i.remoteClients = make(map[string]kube.CLIClient)
+	}
+	rawConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(kube.ConfigLoadingRules(*i.kubeconfig), nil).RawConfig()
```

Comment:
Why do we need all of this if we just call newKubeClientWithRevision with each context anyways? Prefer answer to be a comment in the code for future readers

Reply:
The PR was auto-merged. Created https://github.com/istio/istio/pull/53625 to address this. PTAL